src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -334,44 +346,72 @@ private interface ISymbolMatcher
             bool MatchesSymbol(ISymbol symbol);
         }
 
-        public struct SymbolKindOrTypeKind : IEquatable<SymbolKindOrTypeKind>, ISymbolMatcher, IObjectWritable
+        [DataContract]
+        public readonly record struct SymbolKindOrTypeKind : ISymbolMatcher, IObjectWritable
         {
-            public SymbolKind? SymbolKind { get; }
-            public TypeKind? TypeKind { get; }
-            public MethodKind? MethodKind { get; }
+            private const int KindBits = 6;
+
+            private enum SymbolCategory
+            {
+                Other = 1,
+                Type = 2,
+                Method = 3,
+            }
+
+            // highest 2 bits for category, 6 bits for kind
```

Comment:
not super familiar with message pack serialization - is it not possible to just have the type and kind as properties?  Or was there a perf issue / message size problem if we don't pack this into a byte?

Reply:
It seems like unnecessary complexity, with a slight risk of regression if a future maintainer introduces an error in the shifting/masking.